homeassistant/components/rituals_perfume_genie/switch.py

```diff
@@ -0,0 +1,92 @@
+"""Support for Rituals Perfume Genie switches."""
+from datetime import timedelta
+
+from homeassistant.components.switch import SwitchEntity
+
+from .const import DOMAIN
+
+SCAN_INTERVAL = timedelta(seconds=30)
+
+ON_STATE = "1"
+AVAILABLE_STATE = 1
+
+MANUFACTURER = "Rituals Cosmetics"
+MODEL = "Diffuser"
+ICON = "mdi:fan"
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the diffuser switch."""
+    account = hass.data[DOMAIN][config_entry.entry_id]
+    diffusers = await account.get_devices()
+
+    entities = []
+    for diffuser in diffusers:
+        entities.append(DiffuserSwitch(diffuser))
+
+    async_add_entities(entities, True)
+
+
+class DiffuserSwitch(SwitchEntity):
+    """Representation of a diffuser switch."""
+
+    def __init__(self, diffuser):
+        """Initialize the switch."""
+        self._diffuser = diffuser
+
+    @property
+    def device_info(self):
+        """Return information about the device."""
+        return {
+            "name": self._diffuser.data["hub"]["attributes"]["roomnamec"],
+            "identifiers": {(DOMAIN, self._diffuser.data["hub"]["hublot"])},
+            "manufacturer": MANUFACTURER,
+            "model": MODEL,
+            "sw_version": self._diffuser.data["hub"]["sensors"]["versionc"],
+        }
+
+    @property
+    def unique_id(self):
+        """Return the unique ID of the device."""
+        return self._diffuser.data["hub"]["hublot"]
```

Comment:
Does the `hublot` value meet the unique id requirements here https://developers.home-assistant.io/docs/entity_registry_index/#unique-id-requirements?

Reply:
I have checked multiple devices, they are unique. The code is written at the bottom with a QR code. This QR code must be scanned with the app during setup.
It is possible to set up the integration multiple times with the same account if capital letters are used. Then the ids aren't unique. I will fix this in the next commit.